File: src/main/java/io/polygloat/dtos/request/LanguageDTO.java
package io.polygloat.dtos.request;

import io.polygloat.model.Language;
import lombok.*;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

@Builder
@AllArgsConstructor
@NoArgsConstructor
public class LanguageDTO {
    @Getter
    private Long id;

    @Getter
    @Setter
    @NotBlank
    @Size(max = 100)
    private String name;

    @Getter
    @Setter
    @NotBlank
    @Size(max = 20)
    @Pattern(regexp = "^[^,]*$", message = "can not contain coma")
    private String abbreviation;

    public LanguageDTO(@NotBlank String name, @NotBlank String abbreviation) {
        this.name = name;
        this.abbreviation = abbreviation;
    }

    public static LanguageDTO fromEntity(Language language) {
        LanguageDTO languageDTO = new LanguageDTO(language.getName(), language.getAbbreviation());
        languageDTO.id = language.getId();
        return languageDTO;
    }
}

File: src/main/kotlin/io/tolgee/dtos/request/LanguageDTO.kt
package io.tolgee.dtos.request

import io.tolgee.model.Language
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

data class LanguageDTO(
        var id: Long? = null,

        @field:NotBlank @field:Size(max = 100)
        var name:  String? = null,

        @field:NotBlank @field:Size(max = 20) @field:Pattern(regexp = "^[^,]*$", message = "can not contain coma")
        var abbreviation: String? = null
) {
    companion object {
        fun fromEntity(language: Language): LanguageDTO {
            val languageDTO = LanguageDTO(id = language.id, name = language.name, abbreviation = language.abbreviation)
            languageDTO.id = language.id
            return languageDTO
        }
    }
}